wasm/src/main/java/com/dylibso/chicory/wasm/types/ValType.java

```diff
@@ -229,8 +315,12 @@ public String name() {
 
     public enum TypeIdxCode {
         // heap type
-        EXTERN(0x6F),
-        FUNC(0x70);
+        EXTERN(-17), // 0x6F
+        FUNC(-16), // 0x70
+        EXN(-23), // 0x69
+        BOT(-1),
+        // definedType
+        DEFINED(-2);
```

Comment:
I don't find this `DEFINED` type in the spec, seems like an implementation detail that spills to the user API.
Do you have any reference? Is it necessary or do we have alternative ways to achieve the same?

Reply:
`resolvedFunctionType = null` works for me as a backup option.

The builders plus an explicit resolution phase sounds really appealing though.
I'm not really concerned to deviate from the reference implementation if it's good design and passes the testsuite :-)